Document using extra dicts with pre-commit

danielfdickinson opened the pull request:
In particular, the overrides doc might better be part of the main docs on the website, but I admit to being confused about where and how to contribute to the main docs at the present time.

I'd love to help with them though.

In the meantime I hope you find this documentation effort useful.

I'm happy to make any changes you request. Thank you for great software!

danielfdickinson committed:
Add doc of using overrides with pre-commit

Document using included dictionaries on files for which they are not normally used
(for example, Bash words on Markdown or text files), when using `pre-commit`.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Document using extra dictionaries with pre-commit

This closes #206 and stems from
https://github.com/streetsidesoftware/cspell/discussions/3426
where I had to dig around rather a lot to pull the pieces together.
Hopefully the docs make life easier for the next person who wants
to do this kind of thing.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

Jason3S commented on docs/README-OVERRIDES.md:
````diff
@@ -0,0 +1,78 @@
+
+And the following, as `test1.md` and as `test1.err`:
+
+``` markdown
````
If you want to nest markdown, increase the number of backticks.

Like this:

`````markdown
````markdown
# Nested Markdown

Example Code:
```sh
    if grep -r "something" .; then
        echo "Found"
    fi
```
````
`````

danielfdickinson replied:
Thank you. That's a bit of syntax I didn't know markdown had.

Jason3S commented on docs/README-OVERRIDES.md:
```diff
@@ -0,0 +1,78 @@
+          "dictionaries": [
+              "bash"
+          ],
+          "filename": "**/{*.md,*.txt}"
```
I know it is not alphabetical, but I tend to have `filename` first because it is the `when` clause: _when `filename` matches the glob, apply settings_.

Jason3S replied:
I think it  is clearer if we have the glob apply to one kind of file or directory. Note an array is allowed too.

```js
"filename": ["**/bash_examples/**/*.md", "**/bash_docs/**/*.md"]
```

Jason3S commented on README.md:
````diff
@@ -13,7 +13,26 @@ repos:
 ```
 
-## Setup Custom Dictionary
+## Using additional dictionaries
````
How about a list of **How to Guides**?

How to Guides and Configuration Examples:
- [Use Dictionaries from `cspell-dict`](docs/README-PRE-COMMIT-EXTRA-DICTS.md)
- [Use a Different Dictionary Based Upon the Filename](docs/README-OVERRIDES.md)
- [Example pre-commit setup for French]()

That way the list can grow a bit.

Jason3S commented on docs/README-OVERRIDES.md:
````diff
@@ -0,0 +1,78 @@
+}
+```
+
+And the following, as `test1.md` and as `test1.err`:
````
I find this a bit confusing. I realize you are trying to show the reader how it will impact their spell checking. 
I think it is better to have explicit examples and not have the reader mentally change the name of the file.


- `docs/bash_exmaples/pipes.md` - will use the `bash` dictionary
- `docs/about.md` - will NOT use the `bash` dictionary

Jason3S commented:
@danielfdickinson,

Thank you for giving this a try.

Jason3S commented on docs/README-OVERRIDES.md:
```diff
@@ -0,0 +1,78 @@
+    with)
+        echo "Why?"
+        ;;
+    esac
```
Note: you can add a comment into the doc to ignore issues:

```md
<!--- cspell:ignore esac getopts shopt --->
```

danielfdickinson commented:
Thank you for the review, will update accordingly, soon.

danielfdickinson committed:
Fix nested code block (use Markdown nesting syntax)

Use Markdown nesting syntax rather than indented code block for
for the nested code block.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Ignore other dict words in examples when doing CI

Prevent GitHub workflow (CI) from failing on run of CSpell over the examples
of words not in the default dictionaries, when the workflow is using only
the default dictionaries.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson commented:
I've converted this PR to a draft temporarily as I am making substantial changes based on the review comments (as you will see shortly), and want you to see things as they develop rather than waiting until I've made all my changes.

danielfdickinson committed:
Use an extensible how-to and examples section

Per review comments, tidy up the README references to the how-tos and
configuration examples, and use a title / format to which more docs can
be easily added.

In the process, add a preliminary version of an example setup for French.
Currently a copy of the extra-dics example/how-to but will become a
unique document.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Document file/folder-based overrides

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson commented:
Under docs/ I am considering dropping the `README-` prefix and lower casing the names. Does that make sense to you?

danielfdickinson committed:
Use lowercase filenames in docs folder

And drop the README- prefix. This looks better and is more easily transferred
to the main docs.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Tweak file/folder overrides doc

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Move using a custom dictionary to file under 'docs'

This is more consistent with the other dictionary examples and how-tos.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson commented:
I'm keeping my changes in separate commits to (hopefully) easily revert something that is not wanted.

danielfdickinson committed:
Update documentation examples to cspell v6.7.0

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Update file/folder example to note it adds a dictionary

That it is, the configuration does not disable any dictionaries
that otherwise apply.

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Tweak whitespace

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Improve example setup for French

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson committed:
Add documentation for using `cspell-dict` dictionaries

As noted, this mostly (only?) applies to language dictionaries as other
dictionaries are generally included by default, but only applied to
certain contexts (often a VSCode languageId for the programming language
dictionaries).

Signed-off-by: Daniel F. Dickinson <dfdpublic@wildtechgarden.ca>

danielfdickinson commented:
I'm ready for the next round of comments.

Jason3S commented on docs/pre-commit-example-setup-for-french.md:
````diff
@@ -0,0 +1,148 @@
+
+## Common configuration
+
+### `pre-commit-config.yaml` configuration
+
+Extend the `pre-commit` hook config from the [README.md](../README.md) with
+`additional_dependencies`. For example:
+
+```yaml
+# .pre-commit-config.yaml
+repos:
+  - repo: https://github.com/streetsidesoftware/cspell-cli
+    rev: v6.7.0
+    hooks:
+      - id: cspell
+        additional_dependencies:
+          - "@cspell/dict-fr-fr"
+          - "@cspell/dict-fr-reforme"
+
+```
````
Great!

Jason3S commented on docs/pre-commit-example-setup-for-french.md:
````diff
@@ -0,0 +1,148 @@
+Voici, nous avons les mots française
+```
+
+as again as `mots-française.err`:
````
I don't think you need to repeat the example or show errors.

A most, a directory tree showing which files will use French and which ones will not.

Jason3S commented on docs/pre-commit-example-setup-for-french.md:
```diff
@@ -0,0 +1,148 @@
+When `cspell` is invoked `mots-française.md` should not show errors, but
+`mots-française.err` should.
+
+## Invoking CSpell
```
This could be a list of references to other pages.

Jason3S commented:
@danielfdickinson,

I left a lot of comments. I hope you don't find it too daunting.

Kind regards,
Jason

danielfdickinson commented:
@Jason3S I appreciate the comments quite a lot. I think it is great for getting back into 'real work' mode (I've been off work for a few years due to a major crisis, and your detailed explanations and suggestions are really reminding me how get to the level I want to be at).

danielfdickinson commented:
Sorry for the delay on this; I haven't forgotten, just haven't had the right combination of available time and energy to work on this.

flying-sheep commented:
@Jason3S these docs have already helped me a lot, how about merging them as is and addressing your remaining comments later?

danielfdickinson commented:
@flying-sheep Hi! I'm glad the docs have helped as is :slightly_smiling_face: 

I had quite a bit of 'life stuff' come up, but I also got into what I thought was a shorter term side project that I was focusing on. Since I've realized that it won't be done as quickly as I had expected, I'm planning on scheduling that for a longer term, and making time for this. (Thank you for reminding me of it).

My thought is to make the main doc a reference style. I may also add a small walk-through style, but that remains to be seen.

That said, I wonder if it'd be better to merge as a clearly labelled work in progress, and do pull requests to modify separately rather than to try to have and ideal single pull request. That's up to @Jason3S though.

flying-sheep commented:
Yeah, that’s what I had in mind! Better to get things out there instead of leaving them in a hard to find spot for extended periods of time.

Jason3S commented:
@danielfdickinson,

Thank you.
